### PR TITLE
Add info about cortex-tenant to the docs in Auth guide

### PR DESCRIPTION
**Pull request by blind-oracle**
**What this PR does**:
Adds a section in Auth guide that mentions cortex-tenant proxy that can be used to add tenant headers to Prometheus write requests based on labels.

**Checklist**
- [ ] Tests updated
- [x] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: docs/guides/authentication-and-authorisation.md [hunk 1]
```diff
@@ -30,3 +31,14 @@ should be the same as the one you use to query the data. If they don't match
...
+### Cortex-Tenant
+
+One way to add `X-Scope-OrgID` to Prometheus requests is to use a [cortex-tenant](https://github.com/blind-oracle/cortex-tenant)
+proxy which is able to extract the tenant ID from Prometheus labels.
+
+It can be placed between Prometheus and Cortex and will search for a predefined
+label and use its value as `X-Scope-OrgID` header when proxying the timeseries to Cortex.
+
+This can help to run Cortex in a trusted environment where you want to separate your metrics
+into distinct namespaces by some criteria (teams, applications etc)
```
Comment (pracucci) [inline docs/guides/authentication-and-authorisation.md:44]:
```suggestion
into distinct namespaces by some criteria (eg. teams, applications, etc).
```

File: docs/guides/authentication-and-authorisation.md [hunk 2]
```diff
@@ -30,3 +31,14 @@ should be the same as the one you use to query the data. If they don't match
 you won't see any data. As of now, you can't see series from other tenants.
 
 For more information regarding the tenant ID limits, refer to: [Tenant ID limitations](./limitations.md#tenant-id-naming)
+
+### Cortex-Tenant
+
+One way to add `X-Scope-OrgID` to Prometheus requests is to use a [cortex-tenant](https://github.com/blind-oracle/cortex-tenant)
+proxy which is able to extract the tenant ID from Prometheus labels.
```
Comment (pracucci) [inline docs/guides/authentication-and-authorisation.md:38]: For transparency, I would just explicitly mention that `cortex-tenant` is a community project, not maintained by Cortex maintainers.

**Review by pracucci — APPROVED** [review]
Thanks! LGTM modulo a couple of nits.

File: docs/guides/authentication-and-authorisation.md (hunk 2 above)
Comment (blind-oracle) [reply docs/guides/authentication-and-authorisation.md:38]: Yep, thought about this but wasn't sure how to say this properly. Done.

**Review by khaines — APPROVED** [review]

File: docs/guides/authentication-and-authorisation.md [hunk 3]
```diff
@@ -30,3 +31,16 @@ should be the same as the one you use to query the data. If they don't match
...
+
+It can be placed between Prometheus and Cortex and will search for a predefined
+label and use its value as `X-Scope-OrgID` header when proxying the timeseries to Cortex.
+
+This can help to run Cortex in a trusted environment where you want to separate your metrics
+into distinct namespaces by some criteria (e.g. teams, applications, etc).
+
+Be advised that **cortex-tenant** is a third-party comminity project and it's not maintained by Cortex team.
```
Comment (pracucci) [inline docs/guides/authentication-and-authorisation.md:46]:
```suggestion
Be advised that **cortex-tenant** is a third-party community project and it's not maintained by Cortex team.
```

**Pull request merged**